src/version.cpp

```diff
@@ -63,6 +63,35 @@
 #include <sys/types.h>
 #endif
 
+#ifndef lengthof
+#define lengthof(x) sizeof(x)/sizeof(x[0])
+#endif
+#ifndef _MAX_PATH
+#define _MAX_PATH 512
+#endif
+
+// platform specific support for dumpLibraryInfo
+#if defined(WIN32)
+# include <windows.h>
+# include <psapi.h>
+
+// tell MSVC to link psapi.
+#ifdef  _MSC_VER
+#pragma comment( lib, "psapi" )
```

Comment:
I never used such pragma, but if this is telling to link a library, probably we should this linkage instruction to the CMake code.

Reply:
You're right.  It's a relic.  It's been there for years.  When we had msvc/autotools and CMake build environments, this was a convenient way to specify that link.  However, we should put it into the CMake code.  I'll update the PR to deal with that.

Reply:
Great thanks. If you have difficulties with that I am happy to take care of it.